sound/soc/codecs/rt715-sdw-sdca.c

```diff
@@ -0,0 +1,479 @@
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_MUTE_CTRL, CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_MUTE_CTRL, CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_MUTE_CTRL, CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_MUTE_CTRL, CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_MUTE_CTRL, CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_MUTE_CTRL, CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_VOL_CTRL, CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_VOL_CTRL, CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_VOL_CTRL, CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC8_9_VOL, FU_VOL_CTRL, CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, SMPU_TRIG_ST_EN, SMPU_TRIG_EN_CTRL,
+		CH_00):
+	case RT715_SDCA_ADDR(FUNC_02, SMPU_TRIG_ST_EN, SMPU_TRIG_ST_CTRL,
+		CH_00):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_VOL_CTRL, CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_VOL_CTRL, CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_VOL_CTRL, CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC10_11_VOL, FU_VOL_CTRL, CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC7_27_VOL, FU_MUTE_CTRL, CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC7_27_VOL, FU_MUTE_CTRL, CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC7_27_VOL, FU_VOL_CTRL, CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_ADC7_27_VOL, FU_VOL_CTRL, CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_05):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_06):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_07):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_08):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_05):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_06):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_07):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_08):
+		ret = regmap_read(rt715->sdw_regmap,
+			reg, &sdw_data_0);
+		if (ret < 0)
+			return ret;
+		ret = regmap_read(rt715->sdw_regmap,
+			reg | BIT(13), &sdw_data_1);
+		if (ret < 0)
+			return ret;
+		*val = ((sdw_data_1 & 0xff) << 8) | (sdw_data_0 & 0xff);
+		goto _done_;
+	default:
+		break;
```

Comment:
you can just put the code below up to _done_ here under `default` and remove the `goto`